### ISSUE TITLE
Add mock service as a class

### ISSUE THREAD

**Issue by ne3Vubeki** [getsaf/shallow-render#143]
Hi!
You can add the ability to mock the service as a class. Now you can only mock in the form of an object.
Now I can only mock the service in this way:

```
import {MockService} from "./mocks/mock"

...mock(Service, <Partial<any>>MockService.prototype)
```

But in this case no static methods are passed.

And I would like the whole class to be mocked.

**Comment by getsaf** [comment]
Assuming your `MockService` implements the same (or partial) contract of the original service, you should be able to do this with:

```typescript
import { MockService } from ./mocks/mock;

...mock(Service, new MockService());
```

Also, keep in mind, you may provide any type of mock you wish with `provideMock` (which simply does a `.provide().dontMock()` behind the scenes.

```typescript
// Class Providers
provideMock({provide: Service, useClass: MockService});

// Factory Providers
provideMock({provide: Service, useFactory: () => generateSomeObjectDynamically()});

// Value Providers
provideMock({provide: Service, useValue: new MockService()})
``

**Comment by ne3Vubeki** [comment]
Oh thanks friend!
Didn't find it in the API Doc...

**Comment by ne3Vubeki** [comment]
I tried, but none of the suggested options work. As before, service methods are not seen. In the test, the error: _TypeError: this.service.method is not a function_

**Comment by getsaf** [comment]
Hmmm... Can you please reproduce the issue in the shallow render
stackblitz. And post it here. It'll help me better understand the issue you
are seeing.


https://stackblitz.com/github/getsaf/shallow-render-stackblitz

**Comment by ne3Vubeki** [comment]
https://stackblitz.com/edit/github-nupkzr

**Comment by getsaf** [comment]
Thanks for the StackBlitz example... It looks like we uncovered a bug in `provideMock`. It was not properly "skipping" the built-in mocking when using a Value/Class/Type Provider with `provideMock`.

I just published a fix (v8.5.2) and updated the official StackBlitz with a new example covering the same scenario you have provided.

With 8.5.2, you should be able to use `provideMock` without issue:

```typescript
provideMock({provide: Service, useValue: new MockService()});
```

Thanks for using Shallow, and thanks for reporting this issue!

Please close this issue if this does resolve your problem.

**Comment by ne3Vubeki** [comment]
Thanks!
